### PR TITLE
Break out remote log and stderr handling code to simplify testing

### PR DESCRIPTION
**Pull request by edgewood**
May fix issue 1081, but that issue is hard to reproduce at will, so I can't be sure.  It does remove some needlessly tricky test code.

### CODE REVIEW

File: borg/testsuite/repository.py [hunk 1]
```diff
@@ -440,63 +440,51 @@ def test_crash_before_compact(self):
...
+        self.assert_equal(self.stream.getvalue(), '')
+        # stderr messages don't get an implicit newline
+        self.assert_equal(self.stderr.getvalue(), 'Remote: unstructured stderr message')
 
     def test_old_format_messages(self):
         self.handler.setLevel(logging.DEBUG)
         logging.getLogger().setLevel(logging.DEBUG)
 
-        self.create_repository("$LOG INFO Remote: old format message")
+        handle_remote_line("$LOG INFO Remote: old format message")
```
Comment (ThomasWaldmann) [inline borg/testsuite/repository.py]: maybe use "borg < 1.1" instead of "old".

File: borg/testsuite/repository.py [hunk 2]
```diff
@@ -440,63 +440,51 @@ def test_crash_before_compact(self):
...
         self.assert_equal(self.stream.getvalue(), 'Remote: new format message\n')
+        self.assert_equal(self.stderr.getvalue(), '')
 
     def test_remote_messages_screened(self):
         # default borg config for root logger
         self.handler.setLevel(logging.WARNING)
         logging.getLogger().setLevel(logging.WARNING)
 
-        self.create_repository("$LOG INFO borg.repository Remote: new format info message")
+        handle_remote_line("$LOG INFO borg.repository Remote: new format info message")
```
Comment (ThomasWaldmann) [inline borg/testsuite/repository.py]: "borg >= 1.1 format"

**Comment by ThomasWaldmann** [comment]
my minor nitpick is because of "now" is relative to time of writing, which will be unknown if you look at that in some years.

**Comment by ThomasWaldmann** [comment]
i merged it and will fix the minor nitpicks myself now...

**Comment by edgewood** [comment]
Thanks! Your suggestions did make sense, I just didn't get to them yet.
-- 
Ed Blackman
Sent from my OnePlus One. Please excuse my brevity.


-------- Original Message --------

**Pull request merged**